File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/RawData/Car.cs
namespace RawData
{
    using System.Collections.Generic;
    using System.Linq;

    class Car
    {
        private string model;
        private Engine engine;
        private Cargo cargo;
        private Tire[] tires;

        public Car() { }

        public Car(string model, Cargo cargo, Engine engine, params Tire[] tires)
        {
            this.Model = model;
            this.Cargo = cargo;
            this.Engine = engine;

            this.Tires = new Tire[tires.Length];

            for (int i = 0; i < tires.Length; i++)
            {
                this.Tires[i] = tires[i];
            }
        }

        public Car(string model, Cargo cargo, Engine engine, ICollection<Tire> tires)
        {
            this.Model = model;
            this.Cargo = cargo;
            this.Engine = engine;

            this.Tires = new Tire[tires.Count];
            this.Tires = tires.ToArray();
        }

        public Tire[] Tires
        {
            get { return tires; }
            private set { tires = value; }
        }


        public Cargo Cargo
        {
            get { return cargo; }
            set { cargo = value; }
        }


        public Engine Engine
        {
            get { return engine; }
            set { engine = value; }
        }


        public string Model
        {
            get { return model; }
            set { model = value; }
        }

    }
}

File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Exercise/pr06_raw_data/model/Car.java
package Lesson01DefiningClasses.Exercise.pr06_raw_data.model;

import java.util.ArrayList;
import java.util.List;

public class Car {
    private String model;
    private Engine engine;
    private Cargo cargo;
    private List<Tire> tires;

    public Car(String model,
               int engineSpeed, int enginePower,
               String cargoType, int cargoWeight,
               double[] tiresPressure, int[] tiresAge) {
        this.model = model;
        this.engine = new Engine(engineSpeed, enginePower);
        this.cargo = new Cargo(cargoType, cargoWeight);
        this.tires = new ArrayList<>(4);
        for (int i = 0; i < 4; i++) {
            this.tires.add(new Tire(tiresPressure[i], tiresAge[i]));
        }
    }

    public String getModel() {
        return model;
    }

    public Cargo getCargo() {
        return cargo;
    }

    public Engine getEngine() {
        return engine;
    }

    public List<Tire> getTires() {
        return tires;
    }
}
